src/Avalonia.Controls/TransitioningContentControl.cs

```diff
@@ -136,9 +153,8 @@ private void UpdateContent(bool withTransition)
         var currentPresenter = _isFirstFull ? _presenter2 : Presenter;
 
         if (_lastPresenter != null &&
-            _lastPresenter != currentPresenter &&
-            _lastPresenter.Content == Content)
-            _lastPresenter.Content = null;
+            _lastPresenter != currentPresenter)
+            NotifyOfOldContent(_lastPresenter);
```

Comment:
This will completely remove the transition since the content control has just been setup - this should only be done when the transition is complete.

Reply:
Right, this was my misunderstanding. But I still don't understand why `_lastPresenter.Content == Content` was needed? When is it ever the case?

Reply:
Actually my original design was this in the `ArrangeOverride`:

```cs
        transition.Start(from, to, !IsTransitionReversed, cancel.Token).ContinueWith(x => {
          if (!cancel.IsCancellationRequested) {
            HideOldPresenter();
          }
          else if (from is ContentPresenter oldPresenter) {
            //Even though we are not hiding old presenter yet, we still need to notify that its Content (OldContent) is no longer needed and can be disposed of if need be:
            //NotifyOfOldContent(oldPresenter);
          }
        }, TaskScheduler.FromCurrentSynchronizationContext());
```